### DISCUSSION TITLE
Manual/script-AnimationWindowEvent

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#355] [Notes]
# Manual/script-AnimationWindowEvent

https://docs.unity3d.com/Manual/script-AnimationWindowEvent.html

**Comment by docscomments** [comment]
**Comment by `@HawkDanny`:**  

It is important to note that function called by an animation event is allowed to have a float, string, int, object reference, or an AnimationEvent object as its parameter. However, it **cannot** have a bool as a parameter. Any function with a bool as a parameter will simply not show up in the dropdown of the available functions on the animation event.

**Comment by docscomments** [comment]
**Comment by `@tpryde`:**  

A small gotcha when blending animations is events wont trigger if the animations' blend weight is 50% or less. This is commonly noticed when adding an OnComplete event on the last frame. Consider making a custom StateMachineBehaviour or a coroutine to wait the duration of the animation clip.

**Comment by Syltaris** [comment]
🛠️ Practical Use
You’d write your function to accept an AnimationEvent as its parameter, and then extract the values you need from it. For example:

```csharp
public void HandleEvent(AnimationEvent evt) {
    Debug.Log("Float value: " + evt.floatParameter);
    Debug.Log("String value: " + evt.stringParameter);
    // and so on...
}
```
This lets you pass richer data to your animation-triggered functions without breaking Unity’s one-parameter rule.

**Comment by soraphis** [comment]
> Hover the cursor over a marker to display a tooltip with the function name **and parameter value.**

note that this is not true, it does not show the parameter value
